### ISSUE TITLE
Incompatible with Spring 4.1.2

### ISSUE THREAD

**Issue by tdeo** [jonleighton/spring-watcher-listen#36]
The newest release of Spring (4.1.2) and particularly [this commit](https://github.com/rails/spring/commit/4a5368cc9fee889c32a4a1abaf50dc2672b304e7) makes this watcher broken:

```
NoMethodError: undefined method `synchronize' for #<Spring::Watcher::Listen:0x00000001062a2e50 ...
...

/Users/thierry/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/spring-watcher-listen-2.1.0/lib/spring/watcher/listen.rb:61:in `changed'
/Users/thierry/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/listen-3.8.0/lib/listen/event/config.rb:28:in `call'
/Users/thierry/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/listen-3.8.0/lib/listen/event/processor.rb:117:in `block in _process_changes'
/Users/thierry/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/listen-3.8.0/lib/listen/thread.rb:26:in `rescue_and_log'
/Users/thierry/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/listen-3.8.0/lib/listen/event/processor.rb:116:in `_process_changes'
/Users/thierry/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/listen-3.8.0/lib/listen/event/processor.rb:25:in `block in loop_for'
/Users/thierry/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/listen-3.8.0/lib/listen/event/processor.rb:20:in `loop'
...
```

I believe this call to `synchronize` should be changed to `@mutex.synchronize` to mimic what happened in that commit

**Comment by byroot** [comment]
I just released Spring 4.1.3 which should fix this issue.

**Comment by tdeo** [comment]
Thank you very much for the reactivity 🙏
